Add some demo failure scenarios (#18)
redhat-appstudio/release-service: demos/m4/README.md

The following command can be used to inspect the result of the release:
```bash
oc get release -n demo -o wide
NAME            COMPONENT      SUCCEEDED   REASON      PIPELINERUN         START TIME   COMPLETION TIME   AGE
release-crndf   m4-component   True        Succeeded   m4-strategy-stk9q   41s          32s               50s
```

The important part to notice here is that the Release succeeded and that we keep a reference to the Release PipelineRun.

## Running failure scenarios

There are various checks that must pass in the release process.
The `failures/` directory has scenario files for showing ways in which a Release can fail.

### Missing matching ReleaseLink in managed workspace

A Release cannot be executed unless there is a ReleaseLink in the managed workspace which matches the one in the user workspace.

> Two ReleaseLinks are considered matching if they both specify the same application and the each targets the other's workspace.

Reviewer: ```suggestion
Two ReleaseLinks are considered matching if they both specify the same application and each of them target the other's workspace.
```

Because of this, it marks the Release as `Error` and never starts a Release PipelineRun for it.

To delete the resources used in this scenario, run the following command:
```bash
oc delete -f missing_release_strategy.yaml
```

### Release Pipeline not found

If all the ReleaseLinks are valid and the referenced ReleaseStrategy exists, the last check is that the ReleaseStrategy refers to a valid pipeline.
This scenario shows a Release resource that fails because the ReleaseStrategy in the ReleaseLink refers to a non-existing pipeline.

```bash
$ git clone https://github.com/redhat-appstudio/release-service.git
$ cd release-service/demo/m4/failures
$ oc apply -f missing_release_pipeline.yaml
```

#### What to expect

After running the commands above, you should expect to find:

* a ReleaseLink `test-releaselink-user` in the `releasepipeline-scenario-user` namespace;
* a ReleaseLink `test-releaselink-managed` in the `releasepipeline-scenario-managed` namespace;
* a ReleaseStrategy `test-strategy` in the `releasepipeline-scenario-managed` namespace;
* a Release `test-release` created in the `releasepipeline-scenario-user` namespace;

> * the `REASON` field of the `test-release` Release set to `Error`.

Reviewer: ```suggestion
* the `REASON` field of the `test-release` Release set to `Failed`.
```

The following command can be used to inspect the release:
```bash
oc get release -n releasestrategy-scenario-user -o wide
NAME            COMPONENT      SUCCEEDED   REASON      PIPELINERUN         START TIME   COMPLETION TIME   AGE

> test-release    test           False       Error                                                          29s

Reviewer: Maybe not apply the next one blindly as columns will not align.
```suggestion
test-release    test           False       Failed         test-strategy-d7d69                                             29s
```

```

The controller searches for the `missing-release-pipeline` in the `quay.io/hacbs-release/m4:0.1-alpine` bundle.

> That pipeline does not exist, so the `Release` is marked with `Error`.

Reviewer: ```suggestion
That pipeline does not exist, so the `Release` is marked with `Failed`.
```
